akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -863,6 +864,8 @@ private[akka] class ActorSystemImpl(
   def /(actorName: String): ActorPath = guardian.path / actorName
   def /(path: Iterable[String]): ActorPath = guardian.path / path
 
+  override private[akka] def classicSystem: ActorSystem = this
```

Comment:
classicSystem is defined as internal API in ClassicActorSystemProvider, but is that right? Then it can't be used by libraries? We get away with it in our akka package based projects, but if we start using it like that it should be InternalStableApi.

```
  /** INTERNAL API */
  @InternalApi
  private[akka] def classicSystem: ActorSystem
```

The drawback of making it public is that we expose a somewhat ugly method on ActorSystem.

Reply:
It's only meant to be used by extensions (and indirectly by streams to get to the system materializer extension) so user code should not need to access it, user code can use the typed adapters to go to classic.

Reply:
I think we have expanded on that usage. Right or wrong? https://github.com/akka/akka-persistence-cassandra/blob/master/core/src/main/scala/akka/persistence/cassandra/cleanup/Cleanup.scala#L38

Do you only use it in extensions in Alpakka, @ennru ?

In an extension only this is needed (as Johan said):
```
override def get(system: ClassicActorSystemProvider): Serialization = super.get(system)
```

Reply:
Maybe we'll just mark it as `InternalStableApi`?

Reply:
That's fine by me.

Reply:
I believe it is necessary to make it public API so that libraries not in the `akka` namespace can use it to allow both classic and typed APIs when they in turn use features or libraries that are not adapted yet (eg. Akka 10.1).

Reply:
I defer this discussion to https://github.com/akka/akka/issues/28818